### PR TITLE
[SPARK-41046][CONNECT] Support CreateView in Connect DSL

### PR DESCRIPTION
**Pull request by amaliujia**
### What changes were proposed in this pull request?

This PR supports creating global temp view or local temp view in Connect DSL.

In proto, it is modeled as a command which will be executed immediately on the server side.

### Why are the changes needed?

Improve API coverage.

### Does this PR introduce _any_ user-facing change?

NO

### How was this patch tested?

UT

### CODE REVIEW

**Comment by amaliujia** [comment]
R: @cloud-fan

File: connector/connect/src/main/scala/org/apache/spark/sql/connect/command/SparkConnectCommandPlanner.scala [hunk 1]
```diff
@@ -79,6 +85,32 @@ class SparkConnectCommandPlanner(session: SparkSession, command: proto.Command)
...
+          throw QueryCompilationErrors.invalidViewNameError(createView.getName)
+      }
+
+    val plan = CreateViewCommand(
+      name = tableIdentifier,
+      userSpecifiedColumns = Nil,
+      comment = None,
+      properties = Map.empty,
+      originalText = None,
+      plan = new SparkConnectPlanner(createView.getInput, session).transform(),
```
Comment (zhengruifeng) [inline connector/connect/src/main/scala/org/apache/spark/sql/connect/command/SparkConnectCommandPlanner.scala:105]: `this.transform()` ?

Comment (amaliujia) [reply connector/connect/src/main/scala/org/apache/spark/sql/connect/command/SparkConnectCommandPlanner.scala:105]: In fact it is not...

CommandPlanner and those LogicalPlanPlanner are separated, but Command's input usually are logical plan.

I am thinking to refactor this somehow later to better have those logical contained together.

File: connector/connect/src/main/scala/org/apache/spark/sql/connect/command/SparkConnectCommandPlanner.scala [hunk 2]
```diff
@@ -79,6 +85,32 @@ class SparkConnectCommandPlanner(session: SparkSession, command: proto.Command)
     session.udf.registerPython(cf.getPartsList.asScala.head, udf)
   }
 
+  def handleCreateViewCommand(createView: proto.CreateDataFrameViewCommand): Unit = {
+    val viewType = if (createView.getIsGlobal) GlobalTempView else LocalTempView
+
+    val tableIdentifier =
+      try {
+        session.sessionState.sqlParser.parseTableIdentifier(createView.getName)
```
Comment (cloud-fan) [inline connector/connect/src/main/scala/org/apache/spark/sql/connect/command/SparkConnectCommandPlanner.scala:93]: do we need to parse it? It should be the plain view name.

Comment (amaliujia) [reply connector/connect/src/main/scala/org/apache/spark/sql/connect/command/SparkConnectCommandPlanner.scala:93]: This is from Dataset API implementation and I guess it was to pre-validate if the view name is in a legal format.

If later there is a place to validate it then we can skip this.

Comment (amaliujia) [reply connector/connect/src/main/scala/org/apache/spark/sql/connect/command/SparkConnectCommandPlanner.scala:93]:
```
IDENTIFIER
    : (LETTER | DIGIT | '_')+
    ;
```

The passing in view name might contains more than the defined grammar above

Comment (cloud-fan) [reply connector/connect/src/main/scala/org/apache/spark/sql/connect/command/SparkConnectCommandPlanner.scala:93]: I see

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Review by HyukjinKwon — COMMENTED** [review]
LGTM2

**Pull request closed without merging**